### ISSUE TITLE
Docs reorg - Developers section: create PR for new structure

### ISSUE THREAD

**Issue by dawnkelly09** [filecoin-project/filecoin-docs#1263]
Child of #1254 

- [x] Create a PR to the `feat/developers-section-reorg` branch that stubs out the new structure. New top-level section called "Developers" & subsections as described in this issue.

### Section outline

- Developers
  - Getting started 
    - Hello World tutorial link
    - IPFS & Filecoin Application Design link
  - Overview 
    - What Filecoin brings to applications 
    - IPFS & Filecoin in application design
    - Using Filecoin in other chains (Polygon, Harmony, etc.)
  - Application data
    - NFT data: Use NFT.storage
    - General application data
      - Storage helpers: Web3.storage, Estuary, etc.
      - Manage your own storage deals with raw libraries: Textile Powergate (choose specific SPs, customize pricing, etc.)
    - Advanced application design: dbs, etc.
  - Build on Filecoin L1 (for wallets, exchanges, devtools)
    - Hello world tutorial (extremely simple version that uses a hosted node, something that can be completed in <15 mins)
    - Comprehensive tutorial (maybe point to longfei's Building with Filecoin API tutorial)
    - Signing libraries, devnets, hosted node providers

**Comment by mishmosh** [comment]
Here is the file to edit: https://github.com/filecoin-project/filecoin-docs/blob/main/config/_default/menus/menus.en.toml

**Comment by dawnkelly09** [comment]
@johnnymatthews I think I need a bit of help here. I'm analyzing this menu file to see where to make my changes. I've figured out which pieces go together at you go from main and then nest your way down. I'm trying to figure out how you get the items that are one layer deeper than what is shown in this toml file. For example, https://docs.filecoin.io/build/overview/ I have Build in the Main layer, the Build layer has Get Building and Tools nested in which I can see in the toml file. However, on the existing web page, Get Building has Overview and Quickstart nested in but these don't exist in the toml file. Where do they come from and how can I edit them? 

toml here: https://github.com/filecoin-project/filecoin-docs/blob/feat/developers-section-reorg/config/_default/menus/menus.en.toml

**Comment by dawnkelly09** [comment]
Based purely upon the headings in the current Google doc, the org heading structure is looking like this. I feel like we probably want to tighten up/change some of these names? If folks want to make suggestions/changes in the Google doc, then I can be responsible for updating this comment to keep them matching. I just want a clean place where we can see them: 

https://docs.google.com/document/d/1er7ogbVN6G_4Wb3p8GRbsX3ZASaB6BuA-7tz-68STzc/edit#heading=h.69wnbvjcco0c

**New Outline: Build (in nav bar)**

Developers Overview
	What Filecoin brings to developers
	Using Filecoin with other chains

Build With Filecoin Storage Helpers
	NFTs
	General application data
	Beyond simple storage

Build Directly on Filecoin
	Hello, world!
	Exploring more: local node Filecoin APIs
	Signing libraries
	Testnets
	Hosted node providers
	Bridges & Oracles

Upcoming Features
	FVM
	Retrieval market

**Comment by longfeiWan9** [comment]
@dawnkelly09 I have few suggestions.
1. I think it is better to go with short titles which are more clear, especially for the first level. Since this section is clearly for developer to build, maybe we can consider：
    - Developers Overview -> Overview
    - Build With Filecoin Storage Helpers -> Storage Helpers
    - Build Directly on Filecoin -> Filecoin Tools or Dev Tools
3. Build Directly on Filecoin session is a bit confusing, because it includes tools, networks and tutorials with random order. Maybe we should reorder it. 
   - Hello, world !
   - Local Node APIs
   - Node providers
   - Signing libraries
   - Testnets
   - ~~Bridges & Oracles~~ (Bridges are not maintained anymore, and Oracle is not highly related)

**Comment by mishmosh** [comment]
@longfeiWan9 makes a great point about keeping these short and legible for the nav sidebar. Further truncations:

What Filecoin brings to developers -> Benefits

> Build Directly on Filecoin -> Filecoin Tools or Dev Tools

"Filecoin Tools" could still be misinterpreted by devs who need storage helpers. I don't have a better suggestion right now though! 

I think the sidebar titles don't have to match the headline in the page body 100%. The nav can have the super-short title, and the page body could have slightly longer ones (eg Storage Helpers / Build with Filecoin Storage Helpers).

**Comment by johnnymatthews** [comment]
> For example, https://docs.filecoin.io/build/overview/ I have Build in the Main layer, the Build layer has Get Building and Tools nested in which I can see in the toml file. However, on the existing web page, Get Building has Overview and Quickstart nested in but these don't exist in the toml file. Where do they come from and how can I edit them? - @dawnkelly09 

So yeah you've got the jist of most of it. The `[[main]]` section in `/config/_default/menus/menus.en.toml` deals with the topbar menu:

<img width="865" alt="Screen Shot 2022-06-27 at 8 40 05 PM" src="https://user-images.githubusercontent.com/9611008/176062619-2ebb4a76-2df2-4d55-9bad-8d1b35b6d9ab.png">

The `[[about]]`, `[[networks]]`, `[[getstarted]]`, `[[store]]`, `[[storageprovider]]`, `[[build]]`, and `[[reference]]` sections all manage the sidebar menu:

<img width="260" alt="image" src="https://user-images.githubusercontent.com/9611008/176062773-2a6aba5b-cd06-422b-9854-892ea2b7f7df.png">

The items _within_ those sections are managed within each file. Let's look at [https://docs.filecoin.io/about-filecoin/what-is-filecoin/](https://docs.filecoin.io/about-filecoin/what-is-filecoin/ as an example. In `/content/en/about-filecoin/what-is-filecoin/index.md`. There's a bunch of stuff at the top of the file (called _front-matter_):

```markdown
---
title: "What is Filecoin"
description: "Explore Filecoin, a peer-to-peer network that stores files, with built-in economic incentives to ensure files are stored reliably over time."
menu:
    about:
        parent: "about-filecoin-basics"
weight: 1
aliases:
    - /about-filecoin
---
```

We're interested in the `menu:` bit:

```markdown
menu:          <-- This says that we're looking at the `menus.en.toml` file.
    about:          <-- This says that this page is within the `[[about]]` sub-section.
        parent: "about-filecoin-basics"          <-- This says that this page is in the `about-filecoin-basics` menu item, within the `[[about]]` subsection.
```

The order in which the pages appear in the sidebar menu is defined by the `weight` variable in the front-matter. For this page it's set to `1`, which means it'll go at the very top.

**Comment by dawnkelly09** [comment]
@mosh please take a look at #1307 and let me know if this is what you were looking for. Thanks!

**Comment by dawnkelly09** [comment]
Closed with PR #1332
